fv3core/stencils/dyn_core.py

```diff
@@ -527,15 +527,16 @@ def __call__(self, state):
                     state.wsd,
                     dt,
                 )
-
-                riem_solver3.compute(
+                riem_solver3 = utils.cached_stencil_class(RiemannSolver3)(
```

Comment:
You shouldn't need the caching. you should be able to move that to __init__

Reply:
good catch, I forgot dyn_core was an abject, even though I'm the one that made it so!